InnerEye/ML/configs/classification/CovidHierarchicalModel.py

```diff
@@ -0,0 +1,241 @@
+
+from InnerEye.ML.scalar_config import ScalarLoss, ScalarModelBase
+from InnerEye.ML.utils.augmentation import ScalarItemAugmentation
+from InnerEye.ML.utils.run_recovery import RunRecovery
+from InnerEye.ML.utils.split_dataset import DatasetSplits
+
+from InnerEye.ML.configs.ssl.CovidContainers import COVID_DATASET_ID
+from InnerEye.Common import fixed_paths as fixed_paths_innereye
+
+
+class CovidHierarchicalModel(ScalarModelBase):
+    """
+    Model to train a CovidDataset model from scratch or finetune from SSL-pretrained model.
+
+    For AML you need to provide the run_id of your SSL training job as a command line argument
+    --pretraining_run_recovery_id=id_of_your_ssl_model, this will download the checkpoints of the run to your
+    machine and load the corresponding pretrained model.
+
+    To recover from a particular checkpoint from your SSL run e.g. "recovery_epoch=499.ckpt" please use hte
+    --name_of_checkpoint argument.
+    """
+    use_pretrained_model = param.Boolean(default=False, doc="TBD")
+    freeze_encoder = param.Boolean(default=False, doc="Whether to freeze the pretrained encoder or not.")
+    name_of_checkpoint = param.String(default=None, doc="Filename of checkpoint to use for recovery")
+    test_set_ids = param.String(default=None,
+                                doc="Name of the csv file in the dataset folder with the test set ids.")
+
+    def __init__(self, covid_dataset_id: str = COVID_DATASET_ID, **kwargs: Any):
+        learning_rate = 1e-5 if self.use_pretrained_model else 1e-4
+        super().__init__(target_names=['CVX03vs12', 'CVX0vs3', 'CVX1vs2'],
+                         loss_type=ScalarLoss.CustomClassification,
+                         class_names=['CVX0', 'CVX1', 'CVX2', 'CVX3'],
+                         max_num_gpus=1,
+                         azure_dataset_id=covid_dataset_id,
+                         subject_column="series",
+                         image_file_column="filepath",
+                         label_value_column="final_label",
+                         non_image_feature_channels=[],
+                         numerical_columns=[],
+                         use_mixed_precision=False,
+                         num_dataload_workers=12,
+                         multiprocessing_start_method=MultiprocessingStartMethod.fork,
+                         train_batch_size=64,
+                         optimizer_type=OptimizerType.Adam,
+                         num_epochs=50,
+                         l_rate_scheduler=LRSchedulerType.Step,
+                         l_rate_step_gamma=1.0,
+                         l_rate=learning_rate,
+                         l_rate_multi_step_milestones=None,
+                         **kwargs)
+        self.num_classes = 3
+        if not self.use_pretrained_model and self.freeze_encoder:
+            raise ValueError("You specified training from scratch but you asked to freeze the encoder.")
```

Comment:
may be clearer to phrase more like "No encoder to freeze when training from scratch"